applications/DEM_application/python_scripts/main_script.py

```diff
@@ -8,12 +8,18 @@
 
 # Import MPI modules if needed. This way to do this is only valid when using OpenMPI. For other implementations of MPI it will not work.
 if "OMPI_COMM_WORLD_SIZE" in os.environ or "I_MPI_INFO_NUMA_NODE_NUM" in os.environ:
```

Comment:
my suggestion is not to query for theses env variables, or at least to hide this detection in a different script.

the point is that these variables are MPI-implementation specific (OpenMPI i guess) so that if for example you use mpich or cray mpi instead this would not work

Reply:
One of them is for OpenMPI and the other one is for Intel MPI. We are aware that this is not a good way to detect MPI runs (it is commented one line above), we adopted this way temporarily until some better solution comes up. I know that there are plans to create an MPI kernel, and we can also derive this script into an MPI version of itself. We will take a decision eventually. Thanks for the comment.

Reply:
ok